src/Unison/Server/Share/Definitions.hs

```diff
@@ -186,16 +187,19 @@ mkDefinitionsForQuery nameSearch query = do
           SR.Tp' _ r _ -> Just r
           _ -> Nothing
 
-termDisplayObjectByName :: NameSearch (PG.Transaction e) -> Name -> CodebaseM e (Maybe (TermReference, DisplayObject (Type Symbol Ann) (Term Symbol Ann)))
+-- Nothing means not found
+-- Just Left means constructor
+-- Just Right means term
```

Comment:
Might be worth defining an ADT for this :)

Reply:
I started on this, but the `Nothing` is useful because we do a lot of `MaybeT` and `whenNothingM` with the result. So that leaves "left means constructor, right means term" for putting into an ADT – I think we could probably use this as a generic thing in the Unison codebase, like an even more generic version of `Referent'`:

```
data ConstructorOrTerm a = Constructor a | Term b
```

But I'm not sure what a good name might be. We could leave this alone for now and refactor into a better type later?

Reply:
Realistically never never comes :P But I'm fine leaving it as-is if it doesn't feel right 👍